server/parsec/asgi/rpc.py

```diff
@@ -876,3 +894,86 @@ async def _run_session(self, scope, receive, send) -> None:
                         api_version=self.settled_api_version,
                         **self.headers,
                     )
+
+
+@rpc_router.post("/authenticated/{raw_organization_id}/tos")
+async def tos_api(raw_organization_id: str, request: Request) -> Response:
+    backend: Backend = request.app.state.backend
+
+    parsed = _parse_auth_headers_or_abort(
+        headers=request.headers,
+        raw_organization_id=raw_organization_id,
+        with_authenticated_headers=True,
+        with_invited_headers=False,
+        with_sse_headers=False,
+        expected_accept_type=None,
+        expected_content_type=CONTENT_TYPE_MSGPACK,
+    )
+    assert parsed.authenticated_token is not None
+
+    body: bytes = await _rpc_get_body_with_limit_check(request)
+    outcome = await backend.auth.authenticated_auth(
+        now=DateTime.now(),
+        organization_id=parsed.organization_id,
+        token=parsed.authenticated_token,
+        allow_not_accepted_tos=True,
+    )
+    match outcome:
+        case AuthenticatedAuthInfo() as auth_info:
+            pass
+        case AuthAuthenticatedAuthBadOutcome.ORGANIZATION_EXPIRED:
+            _handshake_abort(
+                CustomHttpStatus.OrganizationExpired, api_version=parsed.settled_api_version
+            )
+        case AuthAuthenticatedAuthBadOutcome.ORGANIZATION_NOT_FOUND:
+            _handshake_abort(
+                CustomHttpStatus.OrganizationNotFound,
+                api_version=parsed.settled_api_version,
+            )
+        case AuthAuthenticatedAuthBadOutcome.TOKEN_TOO_OLD:
+            _handshake_abort(
+                CustomHttpStatus.TokenExpired,
+                api_version=parsed.settled_api_version,
+            )
+        case (
+            AuthAuthenticatedAuthBadOutcome.DEVICE_NOT_FOUND
+            | AuthAuthenticatedAuthBadOutcome.INVALID_SIGNATURE
+        ):
+            _handshake_abort(
+                CustomHttpStatus.BadAuthenticationInfo,
+                api_version=parsed.settled_api_version,
+            )
+        case AuthAuthenticatedAuthBadOutcome.USER_REVOKED:
+            _handshake_abort(
+                CustomHttpStatus.UserRevoked,
+                api_version=parsed.settled_api_version,
+            )
+        case AuthAuthenticatedAuthBadOutcome.USER_FROZEN:
+            _handshake_abort(
+                CustomHttpStatus.UserFrozen,
+                api_version=parsed.settled_api_version,
+            )
+        case AuthAuthenticatedAuthBadOutcome.USER_MUST_ACCEPT_TOS:
+            assert False, "Code should be unreachable !"
```

Comment:
:+1: done.

Reply:
@touilleMan In addition to the suggestion that you applied, I suggested that you add a comment about WHY is unreachable:

> Maybe add a comment here explaining why? Is it because we are in the tos cmds family and the authenticated user must have accepted the TOS to get here?
> 
> Couldn't be a case in which the authenticated user has accepted the TOS, but they changed in the meantime (while being authenticated)?

Reply:
:+1: done.